docs/data/sdks/javascript/index.md

```diff
@@ -983,23 +983,80 @@ var exitCallback = function {
 amplitude.getInstance().init('API_KEY', 'USER_ID', { onExitPage: exitCallback });
 ```
 
-### Custom device IDs
+### Device ID lifecycle
 
-By default, device IDs are randomly generated UUIDs. You can define a custom device ID by setting it as a configuration option or by calling `setDeviceId`.
+By default Amplitude Javascript SDK uses a base64 ID as device ID in order to have more compact string IDs. A 22-character base64 ID has the same range as a 128-bit UUID, which is represented as a 36 character string.
+
+The SDK initializes the device ID in the following order, with the device ID being set to the first valid value encountered:
+
+1. Device id in configuration on initialization
+2. "amp_device_id" value from URL param if `configuration.deviceIdFromUrlParam` is true. Refer to [cross domain tracking](.#cross-domain-tracking-javascript) for more details
+3. Device id in cookie storage. Refer to [cookie management](./#cookie-management) for more details
+4. A randomly generated base64 ID
+
+#### When does a device ID change?
+
+A device ID changes in many scenarios:
+
+- `setDeviceId()` is called explicitly.
+- By default the SDK stores device IDs in cookies, so a device ID will change if a user clears cookies, use another device, or use privacy mode.
+- On initilization, a device ID is passed in from URL param `amp_device` when `deviceIdFromUrlParam` is enabled.
+
+!!!Note "Amplitude Analytics SDKs share an identity store with Experiment SDKs"
+    `setDeviceId` also updates the identity store to propagate new user info to experiment SDK and trigger a fetch if device ID has changed.
+
+#### Set device ID
+
+By default, device IDs are randomly generated base64 IDs. You can define a custom device ID by setting it as a configuration option or by calling `setDeviceId`.
 
 ```js
 amplitude.getInstance().setDeviceId('DEVICE_ID');
 ```
 
-You can retrieve the device ID that Amplitude uses with `Amplitude.getInstance().getDeviceId().` This method can return `null` if a `deviceId` hasn't been generated yet.
-
 !!!note
     Amplitude doesn't recommend defining your own device IDs unless you have your own system for tracking user devices. Make sure the `deviceId` you set is unique to prevent conflicts with other devices in your Amplitude data.
-     It's best practice to use something like a UUID.
+    It's best practice to use something like a UUID.
 
     [See an example](https://github.com/amplitude/Amplitude-Javascript/blob/master/src/uuid.js) of how to generate UUIDs with JavaScript.
 
---8<-- "includes/abbreviations.md"
+#### Get device ID
+
+You can retrieve the device ID that Amplitude uses with `Amplitude.getInstance().getDeviceId()` This method can return `null` if a `deviceId` hasn't been generated yet.
+
+#### Share current device ID to another instance
+
+Sometimes you have more than one Amplitude Javascript SDK instance setup and want to share the device ID across instances. To do so, here are two ways:
```

Comment:
three ways?

Reply:
Delete this line "To do so, here are two ways:" to be more general so that it can be put into a include file.